File: app/src/main/java/com/sample/dhruvmj/waveapp/network/WaveAPI.java
package com.sample.dhruvmj.waveapp.network;

import com.sample.dhruvmj.waveapp.model.Product;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Path;

/**
 * @author Dhruv
 */

public interface WaveAPI {
    String ACCESS_TOKEN = "6W9hcvwRvyyZgPu9Odq7ko8DSY8Nfm";

    /**
     * This method will make RestAPI call to obtain
     * list of products for a certain businessId
     *
     * @param businessId This is the business id for which
     *                   we want to obtain product list
     */
    @Headers("Authorization: Bearer " + ACCESS_TOKEN)
    @GET("businesses/{business_id}/products/")
    Call<List<Product>> getListOfProducts(@Path("business_id") String businessId);
}

File: app/src/main/java/android/sample/wavechallenge/api/WaveApi.kt
package android.sample.wavechallenge.api

import android.sample.wavechallenge.model.Product
import kotlinx.coroutines.Deferred
import retrofit2.http.GET
import retrofit2.http.Path

//A retrofit Network Interface for the Wave Api
interface WaveApi{
    @GET("businesses/{business_id}/products/")
    fun getProductsAsync(@Path("business_id") businessId: String): Deferred<List<Product>>
}